### PR TITLE
Issue #13328: kill mutations in MatchXpathCheck

### PR DESCRIPTION
**Pull request by rdiachenko**
Issue #13328

Attempt to improve this PR: https://github.com/checkstyle/checkstyle/pull/13242

### CODE REVIEW

**Comment by rdiachenko** [comment]
Github, generate report

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/MatchXpathCheck.java [hunk 1]
```diff
@@ -135,7 +135,7 @@ public boolean isCommentNodesRequired() {
 
     @Override
     public void beginTree(DetailAST rootAST) {
-        if (xpathExpression != null) {
+        if (!query.isEmpty()) {
```
Comment (rdiachenko) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/MatchXpathCheck.java:138]: `xpathExpression` depends on `query`, if `query` is empty the `xpathExpression` will always be null.
Updating this condition allows to use default value for `query` with pitest mutation killed.

Comment (romani) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/MatchXpathCheck.java:138]: Ok to merge

**Review by romani — APPROVED** [review]
Ok to merge.
Regression report is not required.

**Review by Vyom-Yadav — APPROVED** [review]
LGTM!

**Pull request merged**